berkeley-mobile/Calendar/Campus/Cells/CampusEventCellView.swift

```diff
@@ -0,0 +1,139 @@
+//
+//  CampusEventCellView.swift
+//  berkeley-mobile
+//
+//  Created by Shawn Huang on 10/3/20.
+//  Copyright © 2020 ASUC OCTO. All rights reserved.
+//
+
+import UIKit
+
+fileprivate let kCardPadding: UIEdgeInsets = UIEdgeInsets(top: 10, left: 15, bottom: 15, right: 10)
+fileprivate let kViewMargin: CGFloat = 5
+
+/// Contained view to be used in CollectionView and TableView cells for campus-wide events
+class CampusEventCellView: UIView {
+
+    init(upcoming: Bool = false) {
+        super.init(frame: .zero)
+        self.translatesAutoresizingMaskIntoConstraints = false
+        self.layoutMargins = kCardPadding
+        
+        if upcoming {
+            timeLabel.textColor = Color.blueText
+        } else {
+            timeLabel.textColor = Color.blackText
+        }
+        
+        self.addSubview(leftVerticalStack)
+        self.addSubview(cellImage)
+        
+        leftVerticalStack.leftAnchor.constraint(equalTo: self.layoutMarginsGuide.leftAnchor).isActive = true
+        leftVerticalStack.topAnchor.constraint(equalTo: self.layoutMarginsGuide.topAnchor).isActive = true
+        leftVerticalStack.rightAnchor.constraint(equalTo: cellImage.leftAnchor, constant: -10).isActive = true
+        leftVerticalStack.bottomAnchor.constraint(equalTo: self.layoutMarginsGuide.bottomAnchor).isActive = true
+        
+        cellImage.rightAnchor.constraint(equalTo: self.layoutMarginsGuide.rightAnchor).isActive = true
+        cellImage.topAnchor.constraint(equalTo: self.layoutMarginsGuide.topAnchor).isActive = true
+        cellImage.bottomAnchor.constraint(equalTo: self.layoutMarginsGuide.bottomAnchor).isActive = true
+        cellImage.widthAnchor.constraint(equalTo: self.widthAnchor, multiplier: 0.35).isActive = true
+        
+        leftVerticalStack.addArrangedSubview(nameLabel)
+        leftVerticalStack.addArrangedSubview(timeLabel)
+        // must constrain heights to prevent name label from taking up all the space
+        timeLabel.heightAnchor.constraint(equalToConstant: 14).isActive = true
+        locationLabel.heightAnchor.constraint(equalToConstant: 14).isActive = true
```

Comment:
This is fine, but btw you can also do this with `.setContentCompressionResistancePriority(.required, for: .vertical)`